match/Repetitions.roc

```diff
@@ -0,0 +1,80 @@
+interface Repetitions
+    exposes [
+        match,
+    ]
+    imports []
+
+match : Str, Str -> Bool
+match = \pattern, text ->
+    # convert pattern and text from Str to List U8
+    patternList = Str.toUtf8 pattern
+    textList = Str.toUtf8 text
+
+    # if the pattern begins with a ^, we'll only check for matches at the beginning of `text`
+    if List.first patternList == Ok '^' then
+        matchHere (List.dropFirst patternList 1) textList
+    else
+        # we'll always need to consider at least 0 as a starting point since text may be empty
+        startingPoints = List.range { start: At 0, end: At (Num.max 1 (List.len textList)) }
```

Comment:
Is `Num.max 1 (List.len textList)` necessary? This generates `[0,1]` when the input is the empty list.

Reply:
I agree this doesn't feel good and I'd like another way to write it. That said, we need the `At:` to be at least one (or otherwise restructure the code) so it's evaluated at least once